chronos/asyncloop.nim

```diff
@@ -1139,7 +1133,7 @@ proc wait*[T](fut: Future[T], timeout = InfiniteDuration): Future[T] =
             clearTimer(timer)
 
           if fut.failed():
-            retFuture.fail(fut.error)
+            retFuture.fail(internalError(fut))
```

Comment:
I'm confused about what `internalError` means...

Reply:
`internal*` provide public access to the otherwise private fields - since currently, all the fields are public it leads to people being able to set the fields "manually" outside of chronos' control which is difficult to provide backwards-compatibility for.